tests/TypeTests.hs

```diff
@@ -11,8 +11,15 @@ import GraphQL.Schema
 
 -- Examples taken from the spec
 
+data DogCommandEnum = Sit | Down | Heel
+
+instance GraphQLEnum DogCommandEnum where
+  enumValues = ["SIT", "DOWN", "HEEL"]
+  enumToValue _ = undefined
+  enumFromValue _ = undefined
+
 -- Alternative might be a sum type with deriving Generic and 0-arity constructors?
-type DogCommand = Enum "DogCommand" '["SIT", "DOWN", "HEEL"]
+type DogCommand = Enum "DogCommand" DogCommandEnum
```

Comment:
This is an unfortunate change ergonomically, but I can see why it might be beneficial for us figuring out a good internal model. No change required.